cli_tools/common/utils/logging/firelog_logger.go

```diff
@@ -0,0 +1,311 @@
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package logging
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	daisyutils "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/daisy"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+	"github.com/google/uuid"
+	"github.com/minio/highwayhash"
+)
+
+var (
+	httpClient            httpClientInterface = &http.Client{Timeout: 5 * time.Second}
+	serverURL                                 = serverURLProd
+	serverLogEnabled                          = false
+	logMutex                                  = sync.Mutex{}
+	nextRequestWaitMillis int64
+)
+
+// constants used by logging
+const (
+	serverURLProd        = "https://firebaselogging-pa.googleapis.com/v1/firelog/legacy/log"
+	key                  = "Ix3TgeJBsH2TH8FUg9_ukW6I6U0t1zOmCySazIA"
+	ImageImportAction    = "ImageImport"
+	ImageExportAction    = "ImageExport"
+	InstanceImportAction = "InstanceImport"
+	targetSizeGb         = "target-size-gb"
+	sourceSizeGb         = "source-size-gb"
+	statusStart          = "Start"
+	statusSuccess        = "Success"
+	statusFailure        = "Failure"
+)
+
+type logResult string
+
+const (
+	failedOnCreateRequest          logResult = "FailedOnCreateRequest"
+	failedOnCreateRequestJSON      logResult = "FailedOnCreateRequestJSON"
+	failedToParseResponse          logResult = "FailedToParseResponse"
+	failedAfterRetry               logResult = "FailedAfterRetry"
+	failedOnUndefinedResponse      logResult = "FailedOnUndefinedResponse"
+	failedOnMissingResponseDetails logResult = "FailedOnMissingResponseDetails"
+	serverLogDisabled              logResult = "ServerLogDisabled"
+)
+
+type httpClientInterface interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+func reverse(str string) string {
+	l := len(str)
+	revStr := make([]byte, l)
+	for i := 0; i <= l/2; i++ {
+		revStr[i], revStr[l-1-i] = str[l-1-i], str[i]
+	}
+	return string(revStr)
+}
+
+// FirelogLoggerInterface is server logger abstraction
+type FirelogLoggerInterface interface {
+	logStart()
+	logSuccess(w *daisy.Workflow)
+	logFailure(err error, w *daisy.Workflow)
+}
+
+// FirelogLogger is responsible for logging to firelog server
+type FirelogLogger struct {
+	ServerURL string
+	ID        string
+	Action    string
+	Params    InputParams
+}
+
+// NewFirelogLogger creates a new server logger
+func NewFirelogLogger(action string, params InputParams) *FirelogLogger {
+	return &FirelogLogger{
+		ServerURL: serverURL,
+		ID:        uuid.New().String(),
+		Action:    action,
+		Params:    params,
+	}
+}
+
+// logStart logs a "start" info to server
+func (l *FirelogLogger) logStart() (*ComputeImageToolsLogExtension, logResult) {
```

Comment:
logStart() sounds like it starts logging, when it's actually logging the start of the operation. Consider renaming.